packages/dds/tree/src/simple-tree/schemaTypes.ts

```diff
@@ -48,6 +48,48 @@ export function isTreeNodeSchemaClass<
  */
 export type AllowedTypes = readonly LazyItem<TreeNodeSchema>[];
 
+/**
+ * Additional information to provide to a Node Schema.
+ *
+ * @typeParam TCustomMetadata - Custom metadata properties to associate with the Node Schema.
+ * See {@link NodeSchemaMetadata.custom}.
+ *
+ * @public
+ */
+export interface NodeSchemaProps<TCustomMetadata = unknown> {
+	/**
+	 * Optional metadata to associate with the Node Schema.
+	 * @remarks Note: this metadata is not persisted in the document.
+	 */
+	readonly metadata?: NodeSchemaMetadata<TCustomMetadata>;
+}
+
+/**
+ * Metadata associated with a Node Schema.
+ *
+ * @remarks Specified via {@link NodeSchemaProps.metadata}.
+ *
+ * @sealed
+ * @public
+ */
+export interface NodeSchemaMetadata<TCustomMetadata = unknown> {
+	/**
+	 * User-defined metadata.
+	 */
+	readonly custom?: TCustomMetadata;
```

Comment:
nitpick: `custom` -> `customData` or some kind of descriptive word `customXXX`

Reply:
Personally, when I see NodeSchemaMetadata.custom I think that's sufficient. It's in a metadata object, so we already know this is some custom subset of metadata from context.

Its subjective though: I'll leave the call up to API review (Which may or may not have already been done for this)

Reply:
Possibly, though I think the context of being under a "Metadata" object is probably sufficient. The standard flow for populating this looks like:

```typescript
schemaFactory.object(..., {
    metadata: {
        ...
        custom: { ... }
    }
})
```

I don't have a particularly strong opinion either way here, though `custom` is what we used for `FieldSchemaMetadata`, which already exists, so I do think we should be consistent.

That said, no release has shipped yet with `FieldSchemaMetadata`, so if you (or anyone) feels strongly about the naming, we do have time to make updates.

Reply:
+1 to "things that are already scoped under some kind of namespace can and should have the shorter name that doesn't repeat the namespace"